plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2209,7 +2209,12 @@ public void createView(ConnectorSession session, SchemaTableName viewName, Conne
         Optional<Table> existing = metastore.getTable(identity, viewName.getSchemaName(), viewName.getTableName());
         if (existing.isPresent()) {
             if (!replace || !isPrestoView(existing.get())) {
-                throw new ViewAlreadyExistsException(viewName);
+                if (TableType.VIRTUAL_VIEW.name().equals(existing.get().getTableType())) {
```

Comment:
IMO we should move this logic to the `CreateViewTask` and handle it in a similar way e.g. `RenameViewTask` handles it (including a check for materialized view). It would work then for all connectors (looks like iceberg connector may have the same issue) + would support MVs.

Reply:
Yes, I also think it should work for all connectors, so the one that really needs to be enhanced is CreateViewTask. So I will resubmit a pr to make an enhancement to CreateViewTask, can I close this pr now?

Reply:
I think it's ok to make changes in this PR (you can squash commits if you need to)

Reply:
cc @lukasz-stec , I am very confused about the function of ConnectorMetadata#getTableHandle in hive.it should return null when the relation is views (just like the comment says in this function). But if I create a view, it will not return null in hive because of tableActions will add a table when view create

Reply:
yeah, hive connector implementation of `ConnectorMetadata#getTableHandle` breaks the contract. I don't think it's that easy to fix though (some features may depend on this behavior)

Reply:
IMO we can first judge whether the view exists according to Metadata#isView, if the view does not exist, then judge whether the table exists, we also need to verify whether this is feasible on other connectors. Do you think we need open other issus about hive connector implementation of ConnectorMetadata#getTableHandle breaks the contract.

Reply:
> IMO we can first judge whether the view exists according to Metadata#isView

yes + `metadata.isMaterializedView` to check if its MV
> Do you think we need open other issus about hive connector implementation of ConnectorMetadata#getTableHandle breaks the contract.

Good question. There is this ticket https://github.com/trinodb/trino/issues/8349 which is a kind of prerequisite for fixing the hive connector but maybe we need another one.

Reply:
Hello, @lukasz-stec .Could you review my code again？ Thank you！